src/engine/qml.js

```diff
@@ -157,7 +157,8 @@ function mergeObjects(obj1, obj2) {
   return mergedObject;
 }
 
-const perContextConstructors = {};
+var importContextIds = 0;
+const perImportContextConstructors = {};
```

Comment:
Yes, I'm also thinking about moving this, perhaps even to properties.
`src/engine/qml.js` has some issues, and perhaps it should be even split or renamed.